tensorflow/core/lib/io/inputstream_interface.h

```diff
@@ -38,7 +38,7 @@ class InputStreamInterface {
   // Typical return codes:
   //  * OK - in case of success.
   //  * OUT_OF_RANGE - not enough bytes remaining before end of file.
-  virtual Status SkipNBytes(int64 bytes_to_skip);
+  virtual Status SkipNBytes(int64 bytes_to_skip) = 0;
```

Comment:
I don't understand the need to make this pure virtual... just keep it virtual and have RandomInputStream override with a more efficient implementation. Then you don't have to copy the implementations in the other subclasses.